spec/controllers/concerns/form_attachment_create_spec.rb

```diff
@@ -43,14 +46,26 @@
       end
     end
 
-    def expect_form_attachment_creation(req_params:)
+    it 'requires file_data to be a file' do
+      params = { param_namespace => { file_data: 'not_a_file_just_a_string' } }
+      post(:create, params: params)
+      expect(response).to have_http_status(400)
+      response_body_errors = JSON.parse(response.body)['errors']
+
+      expect(response_body_errors.size).to eq(1)
+      expect(response_body_errors[0]).to eq(
+        'title' => 'Invalid field value',
+        'detail' => '"String" is not a valid value for "file_data"',
+        'code' => '103',
+        'status' => '400'
+      )
+    end
+
+    def expect_form_attachment_creation
       form_attachment = build(attachment_factory_id, guid: form_attachment_guid)
 
       expect(form_attachment_model).to receive(:new) do
-        expect(form_attachment).to receive(:set_file_data!).with(
```

Comment:
@kevinmirc Mind if I delete this expectation?  `req_params[param_namespace][:file_data]` is a ` Rack::Test::UploadedFile` but `filtered_params[:file_data]` is a `ActionDispatch::Http::UploadedFile`

Reply:
That's fine by me. If I can think of a better way to set that expected param, I'll just do a follow-up PR for it later. Thanks!